extension/src/prototypes/view/containers/ScreenSpaceCamera.tsx

```diff
@@ -0,0 +1,100 @@
+import { useAtomValue } from "jotai";
+import { useCallback, useEffect, useMemo } from "react";
+
+import { ScreenSpaceCameraControllerOptions } from "../../../shared/reearth/types";
+import { useFrame } from "../hooks/useFrame";
+import { enableKeyboardCameraControlAtom } from "../states/app";
+
+import { KeyboardHandlers } from "./KeyboardHandlers";
+
+type ScreenSpaceCameraProps = {
+  tiltByRightButton: boolean;
+  minimumZoomDistance?: number;
+  maximumZoomDistance?: number;
+};
+export const ScreenSpaceCamera = ({
+  tiltByRightButton = false,
+  minimumZoomDistance = 1.5,
+  maximumZoomDistance = Infinity,
+}: ScreenSpaceCameraProps) => {
+  const useKeyboard = useAtomValue(enableKeyboardCameraControlAtom);
+
+  const optionsWhenUseKeyboard: ScreenSpaceCameraControllerOptions = useMemo(
+    () => ({
+      zoomEventTypes: [],
+      rotateEventTypes: [],
+      tiltEventTypes: [],
+      lookEventTypes: [
+        "left_drag",
+        {
+          eventType: "left_drag",
+          modifier: "ctrl",
+        },
+        {
+          eventType: "left_drag",
+          modifier: "shift",
+        },
+      ],
+      maximumZoomDistance,
+      minimumZoomDistance,
+      enableCollisionDetection: !useKeyboard,
+    }),
+    [minimumZoomDistance, maximumZoomDistance, useKeyboard],
+  );
+
+  const optionsWhenTiltByRightButton: ScreenSpaceCameraControllerOptions = useMemo(
+    () => ({
+      zoomEventTypes: ["middle_drag", "wheel", "pinch"],
+      rotateEventTypes: ["left_drag"],
+      tiltEventTypes: [
+        "right_drag",
+        "pinch",
+        {
+          eventType: "left_drag",
+          modifier: "ctrl",
+        },
+        {
+          eventType: "right_drag",
+          modifier: "ctrl",
+        },
+      ],
+      maximumZoomDistance,
+      minimumZoomDistance,
+      enableCollisionDetection: !useKeyboard,
+    }),
+    [minimumZoomDistance, maximumZoomDistance, useKeyboard],
+  );
+
+  const cb = useCallback(() => {
+    const cameraPotion = window?.reearth?.camera?.position;
+    const negativePiToPi = window.reearth?.scene?.negativePiToPi(cameraPotion?.roll);
+    console.log(negativePiToPi);
+    if (!cameraPotion || !useKeyboard) return;
+    if (typeof negativePiToPi !== "undefined") {
+      if (Math.abs(negativePiToPi) > Math.PI / 86400) {
+        window.reearth?.camera?.rollCameraHorizontal();
+      }
+    }
```

Comment:
```suggestion
    const negativePiToPi = window.reearth?.scene?.negativePiToPi(cameraPotion?.roll);
    if (!cameraPotion || !useKeyboard || negativePiToPi != null) return;
    if (Math.abs(negativePiToPi) > Math.PI / 86400) {
      window.reearth?.camera?.rollCameraHorizontal();
    }
```

Reply:
Update: we have implemented this logic in reearth.